erdblick_app/app/view.component.ts

```diff
@@ -49,54 +100,163 @@ declare let window: DebugWindow;
     private openStreetMapLayer: ImageryLayer | null = null;
-    private marker: Entity | null = null;
+    private markerCollection: BillboardCollection | null = null;
     private tileOutlineEntity: Entity | null = null;
     menuItems: MenuItem[] = [];
     private cameraIsMoving: boolean = false;
+    private ignoreNextCameraUpdate: boolean = false;
+    is2DMode: boolean;
+    
+    // State to preserve during viewer reinitialization
+    private viewerState: {
+        openStreetMapLayerAlpha: number;
+        openStreetMapLayerShow: boolean;
+        markerPositions: Cartesian3[];
+        tileOutlineEntity: Entity | null;
+        cameraState: any;
+        menuItems: MenuItem[];
+    } | null = null;
 
     /**
      * Construct a Cesium View with a Model.
      * @param mapService The map model service providing access to data
-     * @param styleService
      * @param featureSearchService
      * @param parameterService The parameter service, used to update
      * @param jumpService
+     * @param inspectionService
+     * @param keyboardService
+     * @param menuService
      * @param coordinatesService Necessary to pass mouse events to the coordinates panel
      * @param appModeService
      */
-    constructor(public mapService: MapService,
-                public styleService: StyleService,
-                public featureSearchService: FeatureSearchService,
-                public parameterService: ParametersService,
-                public jumpService: JumpTargetService,
-                public inspectionService: InspectionService,
-                public keyboardService: KeyboardService,
-                public menuService: RightClickMenuService,
-                public coordinatesService: CoordinatesService,
+    constructor(private mapService: MapService,
+                private featureSearchService: FeatureSearchService,
+                private parameterService: ParametersService,
+                private jumpService: JumpTargetService,
+                private inspectionService: InspectionService,
+                private keyboardService: KeyboardService,
+                private menuService: RightClickMenuService,
+                private coordinatesService: CoordinatesService,
                 public appModeService: AppModeService) {
 
+        this.is2DMode = this.parameterService.parameters.getValue().mode2d;
+
         this.mapService.tileVisualizationTopic.subscribe((tileVis: TileVisualization) => {
+            // Safety check: ensure viewer exists and is not destroyed
+            if (!this.viewer || (typeof this.viewer.isDestroyed === 'function' && this.viewer.isDestroyed())) {
+                console.warn('Cannot render tile visualization: viewer not available');
+                return;
+            }
+            
             tileVis.render(this.viewer).then(wasRendered => {
-                if (wasRendered) {
+                if (wasRendered && this.viewer && this.viewer.scene) {
```

Comment:
Aren't the extra checks here redundant, as they were already performed at the beginning of the function?